source/common/upstream/outlier_detection_impl.cc

```diff
@@ -177,6 +198,17 @@ void DetectorHostMonitorImpl::putResultWithLocalExternalSplit(Result result,
 // config parameter.
 void DetectorHostMonitorImpl::putResult(Result result, absl::optional<uint64_t> code) {
   put_result_func_(this, result, code);
+
+  // Call extensions
+  if (monitors_set_.empty()) {
+    return;
+  }
+
+  // Pass the local origin event to all registered extension monitors.
+  // Only monitors "interested" in local origin event will process the result.
+  // Those not "interested" will ignore the call.
+  monitors_set_.forEach(
+      [result](const ExtMonitorPtr& monitor) { monitor->putResult(LocalOriginEvent(result)); });
```

Comment:
I left it as is. There are actually 3 different ways how it is used. `putResult` is called with 2 different types of parameter (`HttpCode` and `LocalOrigin`) plus `reset`. My next step will be to add support for database type errors so most likely `putResult` will be called with yet another type of parameter. I can revisit it then, whether it can be improved and simplified.